src/otx/core/data/pre_filtering.py

```diff
@@ -88,7 +88,15 @@ def remove_unused_labels(dataset: DmDataset, data_format: str, ignore_index: int
         raise ValueError(msg)
     if len(used_labels) == len(original_categories):
         return dataset
+    if data_format == "arrow" and max(used_labels) != len(original_categories) - 1:
+        # we assume that empty label is always the last one. If it is not explicitly added to the dataset,
+        # (not in the used labels) it will be filtered out.
```

Comment:
Can we verify that this assumption is always correct? Is there any code or documentation can reference for this?

Reply:
I will try to find it on the Datumaro side

Reply:
Unfortunately, I haven't found the 100% proof for that. I can only see that labels are sorted in label schema by assigned ID
https://github.com/intel-innersource/frameworks.ai.interactive-ai-workflow.sonoma-creek/blob/386159bd64df784985efb2cc460d9a4b00d63435/src/sc_sdk/sc_sdk/otx/api/entities/label_schema.py#L72
 And Empty label should have the largest ID value.

Reply:
@wonjuleee Can you confirm that we can use this workaround?